packages/schematics/angular/module/index.ts

```diff
@@ -82,18 +152,37 @@ export default function (options: ModuleOptions): Rule {
     options.name = parsedPath.name;
     options.path = parsedPath.path;
 
+    let routingModulePath: Path | undefined;
+    const isLazyLoadedModuleGen = options.route && options.module;
+    if (isLazyLoadedModuleGen) {
+      options.routingScope = RoutingScope.Child;
+      routingModulePath = getRoutingModulePath(host, options);
+    }
+
+    const project = getProject(host, options.project as string);
     const templateSource = apply(url('./files'), [
-      options.routing ? noop() : filter(path => !path.endsWith('-routing.module.ts.template')),
+      options.routing || isLazyLoadedModuleGen && !!routingModulePath
+        ? noop()
+        : filter(path => !path.endsWith('-routing.module.ts.template')),
       applyTemplates({
         ...strings,
         'if-flat': (s: string) => options.flat ? '' : s,
+        lazyRoute: isLazyLoadedModuleGen,
+        routeDeclarationInlined: !routingModulePath,
         ...options,
       }),
       move(parsedPath.path),
     ]);
 
     return chain([
-      addDeclarationToNgModule(options),
+      !isLazyLoadedModuleGen ? addDeclarationToNgModule(options) : noop(),
+      addRouteDeclarationToNgModule(options, project, routingModulePath),
+      isLazyLoadedModuleGen
+        ? schematic('component', {
+            ...options,
+            skipImport: true,
```

Comment:
The lazy loaded component needs to be imported in the lazy loaded module.

Reply:
Ah, I am manually doing this in the template rather than making use of the component schematics functionality. Makes sense.

Reply:
I think this should be done via component schematic.

Reply:
I just pointed that I am doing it the wrong way, apparently. I'll have to keep the import in the routing module though, since we don't need the component added to the declarations array there.